pkg/apis/core/v1/package_webhook.go

```diff
@@ -57,14 +61,30 @@ func (r *Package) ValidateCreate() error {
 	if err != nil {
 		return err
 	}
+
+	// Ensure size limits
+	if len(r.Spec.Source.Literal) > int(ArchiveLiteralSizeLimit) {
+		err := ferror.MakeError(ferror.ErrorInvalidArgument,
+			fmt.Sprintf("Package literal larger than %s", humanize.Bytes(uint64(ArchiveLiteralSizeLimit))))
+		return err
+	}
+	if len(r.Spec.Deployment.Literal) > int(ArchiveLiteralSizeLimit) {
+		err := ferror.MakeError(ferror.ErrorInvalidArgument,
+			fmt.Sprintf("Package literal larger than %s", humanize.Bytes(uint64(ArchiveLiteralSizeLimit))))
```

Comment:
If `Spec.Deployment.Literal` is larger than 256 kb, probably kubernetes wont accept this request. Please verify this once.